File: C++/operator/arithmetic.cpp
using namespace std;
#include <iostream>

void integerArithmetic();
void decimalArithmetic();
void charArithmetic();
void stringArithmetic();

int main()
{
  integerArithmetic();
  decimalArithmetic();
  charArithmetic();
  stringArithmetic();

  return 0;
}

void integerArithmetic()
{
  int x = 5;
  int y = 3;

  std::cout << "5 + 3 = " << x + y << std::endl;
  std::cout << "5 - 3 = " << x - y << std::endl;
  std::cout << "5 * 3 = " << x * y << std::endl;
  std::cout << "5 / 3 = " << x / y << std::endl;
  std::cout << "5 % 3 = " << x % y << std::endl;

  std::cout << "\n"
            << std::endl;
}

void decimalArithmetic()
{
  float x = 5.25;
  float y = 3.3;

  std::cout << "5.25 + 3.3 = " << x + y << std::endl;
  std::cout << "5.25 - 3.3 = " << x - y << std::endl;
  std::cout << "5.25 * 3.3 = " << x * y << std::endl;
  std::cout << "5.25 / 3.3 = " << x / y << std::endl;
  std::cout << "++x = " << ++x << std::endl;
  std::cout << "--x = " << --x << std::endl;

  std::cout << "\n"
            << std::endl;
}

void charArithmetic()
{
  char char1 = 'A';
  char char2 = 'B';

  std::cout << "char1 + char2 = " << int(char1 + char2) << std::endl;
  std::cout << "char1 - char2 = " << int(char1 - char2) << std::endl;
  std::cout << "char1 * char2 = " << int(char1 * char2) << std::endl;
  std::cout << "char1 / char2 = " << int(char1 / char2) << std::endl;

  std::cout << "\n"
            << std::endl;
}

void stringArithmetic()
{
  string firstname = "Teeramate";
  string lastname = "Kantima";

  std::cout << "My name is " << firstname + " " + lastname << std::endl;

  std::cout << "\n"
            << std::endl;
}

File: JavaScript/operator/arithmetic.js
function NumberArithmetic() {
  let x = 10;
  let y = 5;

  console.log("******** Number ********");
  console.log("x + y =", x + y);
  console.log("x - y =", x - y);
  console.log("x * y =", x * y);
  console.log("x ** y =", x ** y);
  console.log("x / y =", x / y);
  console.log("x % y =", x % y);
  console.log("x++ =", x++);
  console.log("x-- =", x--);
  console.log("\n");
}

function DecimalArithmetic() {
  let x = 12.34;
  let y = 2.23;

  console.log("******** Decimal ********");
  console.log("x + y =", x + y);
  console.log("x - y =", x - y);
  console.log("x * y =", x * y);
  console.log("x ** y =", x ** y);
  console.log("x / y =", x / y);
  console.log("x % y =", x % y);
  console.log("x++ =", x++);
  console.log("x-- =", x--);
  console.log("\n");
}

function StringArithmetic() {
  let firstname = "Jack";
  let lastname = "Dawson";

  console.log("******** String ********");
  console.log("firstname + lastname is", firstname + " " + lastname);
  console.log("\n");
}

function MixArithmetic() {
  let number = 10;
  let numStr = "5";
  let str = "hello";

  console.log("******** Mix data type ********");
  console.log("10 + '5' is ", number + numStr); // type is String
  console.log("10 - 'hello' is ", number - str);
  console.log("10 * '5' is ",
              number * numStr); // when use -, *, /, %, ** type is Number
  console.log("10 / '5' is ", number / numStr);
  console.log("10 % '5' is ", number % numStr);
  console.log("10 ** '5' is ", number ** numStr);
  console.log("\n");
}

NumberArithmetic();
DecimalArithmetic();
StringArithmetic();
MixArithmetic();
